source/jobs/JobsFeature.cpp

```diff
@@ -180,6 +195,14 @@ void JobsFeature::subscribeToUpdateJobExecutionStatusRejected(string jobId)
         std::bind(
             &JobsFeature::updateJobExecutionStatusRejectedHandler, this, std::placeholders::_1, std::placeholders::_2),
         std::bind(&JobsFeature::ackSubscribeToUpdateJobExecutionRejected, this, std::placeholders::_1));
+
+    if (std::future_status::timeout == updateRejectedPromise.get_future().wait_for(std::chrono::seconds(10)))
```

Comment:
Can't you use the promise returned by subscribeToUpdateJobExecutionRejected instead?

https://aws.github.io/aws-iot-device-sdk-js-v2/classes/aws_iot_device_sdk.iotjobsclient.html#subscribetoupdatejobexecutionrejected